java-squid/src/main/java/org/sonar/java/se/symbolicvalues/SymbolicValueRelation.java

```diff
@@ -112,9 +142,9 @@ public Boolean impliedBy(Collection<SymbolicValueRelation> knownRelations, Set<S
     List<SymbolicValueRelation> result = new ArrayList<>();
     for (SymbolicValueRelation relation : knownRelations) {
       boolean used = false;
-      if (v1.equals(relation.v1)) {
+      if (leftOp.equals(relation.leftOp)) {
         used = result.addAll(relation.combinedRelations(knownRelations, usedRelations));
-      } else if (v1.equals(relation.v2)) {
+      } else if (leftOp.equals(relation.rightOp)) {
         used = result.addAll(relation.symmetric().combinedRelations(knownRelations, usedRelations));
       }
       if (used) {
```

Comment:
I think I figured it out but I guess this method and this if really deserves a comment.